network-config-analyzer/FWRule.py

```diff
@@ -0,0 +1,357 @@
+        :param is_src: bool flag to indicate if element is src (True) or dst (False)
+        :return: string of the represented element with src or dst description of fields
+        """
+        ns_prefix = 'src_ns: ' if is_src else 'dst_ns: '
+        pods_prefix = ' src_pods: ' if is_src else ' dst_pods: '
+        suffix = ' ' if is_src else ''
+        return ns_prefix + self.get_ns_str() + pods_prefix + self.get_pod_str() + suffix
+
+    def __hash__(self):
+        return hash(str(self))
+
+    def __eq__(self, other):
+        return set(self.ns_info) == set(other.ns_info)
+
+    def is_system_ns(self):
+        """
+        :return: True if this element has one namespace and it ends with "system"
+        """
+        return len(self.ns_info) == 1 and str(list(self.ns_info)[0]).endswith("-system")
+
+    def get_pods_set(self, cluster_info):
+        """
+        :param cluster_info: an object of type ClusterInfo, with relevant cluster topology info
+        :return: a set of pods in the cluster represented by this element
+        """
+        res = set()
+        for ns in self.ns_info:
+            res |= cluster_info.ns_dict[ns]
+        return res
+
+
+class PodElement(FWRuleElement):
+    """
+    This is the class for single pod element in fw-rule
+    """
+
+    def __init__(self, element):
+        """
+        Create a PodElement object
+        :param element: the element of type Pod
+        """
+        super().__init__([element.namespace])
+        self.element = element
+
+    def get_pods_yaml_obj(self):
+        """
+        :return: list[string] for the field src_pods or dst_pods in representation for yaml object
+        """
+        return [str(self.element.owner_name)]
+
+    def get_pod_str(self):
+        """
+        :return: string for the field src_pods or dst_pods in representation for txt rule format
+        """
+        # return '[' + str(self.element) + ']'
+        return '[' + str(self.element.owner_name) + ']'
+
+    def __str__(self):
+        """
+        :return: string of the represented element
+        """
+        return 'ns: ' + self.get_ns_str() + ', pods: ' + self.get_pod_str()
+
+    def get_elem_str(self, is_src):
+        """
+        :param is_src: bool flag to indicate if element is src (True) or dst (False)
+        :return: string of the represented element with src or dst description of fields
+        """
+        ns_prefix = 'src_ns: ' if is_src else 'dst_ns: '
+        pods_prefix = ' src_pods: ' if is_src else ' dst_pods: '
+        suffix = ' ' if is_src else ''
+        return ns_prefix + self.get_ns_str() + pods_prefix + self.get_pod_str() + suffix
+
+    def __hash__(self):
+        return hash(str(self))
+
+    # TODO: should compare to other types as well, and make the comparison based on actual pods set?
+    def __eq__(self, other):
+        return isinstance(other, PodElement) and self.element == other.element and set(self.ns_info) == set(
+            other.ns_info)
```

Comment:
again, why do we need the conversion to set? Also, better call `super().__eq__ ` for this comparison

Reply:
done